model/src/test/test_assert_type.py

```diff
@@ -0,0 +1,41 @@
+# numpydoc ignore=GL08
+
+import numpyro.distributions as dist
+import pytest
+from pyrenew.metaclass import DistributionalRV, RandomVariable, _assert_type
+
+
+def test_valid_assertion_types():
+    """
+    Test valid assertion types in _assert_type.
+    """
+
+    values = [
+        5,
+        "Hello",
+        (1,),
+        DistributionalRV(name="rv", dist=dist.Beta(1, 1)),
+    ]
+    arg_names = ["input_int", "input_string", "input_tuple", "input_rv"]
+    input_types = [int, str, tuple, RandomVariable]
+
+    for arg, value, input in zip(arg_names, values, input_types):
+        _assert_type(arg, value, input)
+
+
+def test_invalid_assertion_types():
+    """
+    Test invalid assertion types in _assert_type.
+    """
+
+    values = [None] * 4
+    arg_names = ["input_int", "input_string", "input_tuple", "input_rv"]
+    input_types = [int, str, tuple, RandomVariable]
+
+    for arg, value, input in zip(arg_names, values, input_types):
+        with pytest.raises(
+            TypeError,
+            match=f"{arg} must be an instance of {input}. "
+            f"Got {type(value)}",
```

Comment:
It's probably enough just to check if it raises a `TypeError`. Duplicating the entire error message seems unnecessary.

Reply:
fixed in https://github.com/CDCgov/multisignal-epi-inference/pull/351/commits/24c15941e49819b13cef04e48ae5341346a1bff3

Reply:
I am weakly in favor of checking the error message but I'll defer to @damonbayer